infrahub_sdk/transforms.py

```diff
@@ -108,7 +120,8 @@ def get_transform_class_instance(
         transform_class = getattr(module, transform_config.class_name)
 
         # Create an instance of the class
-        transform_instance = transform_class()
+        transform_instance = asyncio.run(transform_class.init(client=client, branch=branch))
```

Comment:
We are deprecating the usage of the `init` method on different classes in the SDK. We should probably not use this method any longer (we cannot remove it yet, as the backend Infrahub code is still leveraging it). Instead we should consider changing the __init__ method.

Reply:
Acknowledged. Looking at the method, there is actually no asynchronous code in the async `init` method. The only thing the `init` method adds over `__init__` is the creation of a client.

I added a `@cached_property` (from functools) of `client` that initializes the client in the same way this init method does. The client is currently instantiated synchronously using the BaseClient's `__init__` method, so I believe there should be no issues here.

I then instantiated the `TransformClass` directly using it's __init__ method synchronously so that we don't call that `init` method.

The only oddity here is that the client is a `InfrahubClient` instead of an `InfrahubClientSync`. I added a block of code in the `_run_transform` method to determine which type of object we're dealing with (as the `render` command uses the synchronous version) and run it correctly.